### PR TITLE
Add strict to BaseConfig

### PR DESCRIPTION
**Pull request by adriangb**
Linked to #4664

### CODE REVIEW

File: pydantic/types.py [hunk 1]
```diff
@@ -156,9 +156,9 @@ def confloat(
 
 def conbytes(
     *,
-    min_length: int = None,
-    max_length: int = None,
-    strict: bool = None,
+    min_length: int | None = None,
+    max_length: int | None = None,
+    strict: bool | None = None,
```
Comment (adriangb) [inline pydantic/types.py:161]: Drive by

File: tests/test_strict.py [hunk 2]
```diff
@@ -34,3 +34,43 @@ def test_parse_strict_mode_on_field_invalid(value: Any, ModelWithStrictField: Ty
...
+    return ModelWithStrictConfig
+
+
+def test_parse_model_with_strict_config_enabled(ModelWithStrictConfig: Type[BaseModel]) -> None:
+    with pytest.raises(ValidationError):
+        ModelWithStrictConfig(a='1', b=2, c=3, d=4)
+    with pytest.raises(ValidationError):
+        ModelWithStrictConfig(a=1, b=2, c='3', d=4)
+    with pytest.raises(ValidationError):
+        ModelWithStrictConfig(a=1, b=2, c=3, d='4')
```
Comment (samuelcolvin) [inline tests/test_strict.py:62]: probably good to check the errors are in the right place.

File: tests/test_strict.py [hunk 3]
```diff
@@ -34,3 +34,43 @@ def test_parse_strict_mode_on_field_invalid(value: Any, ModelWithStrictField: Ty
...
+    ModelWithStrictConfig(a=1, b='2', c=3, d=4)
+    ModelWithStrictConfig(a=1, b=2, c=3, d=4)
```
Comment (samuelcolvin) [inline tests/test_strict.py:64]: Probably best to do

```
assert ....dict() == {'a': 1, 'b': 2, 'c': 3, 'd': 4}
```
On all cases.

**Review by samuelcolvin — COMMENTED** [review]
otherwise looking good.

**Comment by samuelcolvin** [comment]
Looking good, feel free to merge when you think this is ready.

**Pull request merged**